baseband/vdif/tests/test_vdif.py

```diff
@@ -664,36 +664,75 @@ def test_stream_writer(self, tmpdir):
             assert np.all(fhns.read() == record)
 
     def test_subset(self, tmpdir):
-        # Test subsetting on a 8 thread, 4 channel dataset to file.
-
-        # Make the file.
-        test_file = str(tmpdir.join('test.vdif'))
+        # Use the default sample file.
         with vdif.open(SAMPLE_FILE, 'rs') as fh:
             data = fh.read()
-            data = np.array([data, abs(data),
-                             -data, -abs(data)]).transpose(1, 2, 0)
-            with vdif.open(test_file, 'ws',
-                           nthread=8, nchan=4, sample_rate=fh.sample_rate,
-                           samples_per_frame=fh.samples_per_frame // 4,
-                           complex_data=fh.complex_data, bps=fh.bps,
-                           edv=fh.header0.edv, station=fh.header0.station,
-                           time=fh.start_time) as fw:
-                fw.write(data)
+            sample_rate = fh.sample_rate
+            samples_per_frame = fh.samples_per_frame
+            header0 = fh.header0
+
+        with vdif.open(SAMPLE_FILE, 'rs', subset=slice(0, 8, 2)) as fhn:
+            assert fhn.sample_shape == (4,)
+            check = fhn.read()
```

Comment:
`test_gsb.py` also uses `np.all(fh.read()...)` when checking subset; I would change it to use `check` there too.